pilot/pkg/model/push_context.go

```diff
@@ -599,6 +599,12 @@ var (
 		"Total virtual services known to pilot.",
 	)
 
+	// totalDestinationRules tracks the total number of destination rules
+	totalDestinationRules = monitoring.NewGauge(
```

Comment:
TBH, i donot know what's the use case about this. Also doubt about the vs monitor. 

If the use case is reasonable, why not we monitor all other kinds of configs.  This is not the business of istio, like when we are using k8s, should k8s provide metrics of how many pods/deployments/secrets. I doubt not.

Reply:
The usecase we ran in to recently is because of a bug in our config generation logic, we accidentally deleted configs. Looking at VS count helped. I was thinking if we have the same for DR, we could alert on if we see drastic decrease in these two resources.

K8s does not provide them AFAIK.

I am fine to drop this if you think it is not needed. But we can not delete VS one though, as people might be using it.

Reply:
I have added for DR because these two are most frequently used.

Reply:
I think this is not core istio's business., but the responsibility of a company monitoring system. Otherwise we could include so many things in istio